### PR TITLE
[Feature] Test batch

### PR DESCRIPTION
**Pull request by kennymckormick**
Add feature `test_batch` for extremely large model testing.
Background: 
For extremely large models, you can not even forward one sample at a time (which includes 10x3=30 views). By setting test_batch in test_cfg, you can forward `test_batch` views at a time.

### CODE REVIEW

File: mmaction/models/recognizers/recognizer3d.py [hunk 1]
```diff
@@ -29,13 +31,28 @@ def _do_test(self, imgs):
...
         imgs = imgs.reshape((-1, ) + imgs.shape[2:])
 
-        x = self.extract_feat(imgs)
-        if hasattr(self, 'neck'):
-            x, _ = self.neck(x)
+        if self.test_batch is not None:
+            tot = imgs.shape[0]
+            assert num_segs == tot, ('test_batch is only compatible with '
+                                     'batch_size == 1')
+            ptr = 0
```
Comment (dreamerlin) [inline mmaction/models/recognizers/recognizer3d.py:38]: `ptr` and `total` are not good names. How about `batch_point` and `total_batches` ?

**Comment by dreamerlin** [comment]
We may need some explanation for this feature in docs.

File: mmaction/models/recognizers/recognizer3d.py (hunk 1 above)
Comment (kennymckormick) [reply mmaction/models/recognizers/recognizer3d.py:38]: I will change it to view_ptr and total_views, I think `ptr` is widely used to represent a pointer.

**Comment by innerlee** [comment]
As discussed, we change the name to `max_testing_views`

**Review by innerlee — APPROVED** [review]

**Pull request merged**